pkg/daemon/update.go

```diff
@@ -423,10 +427,32 @@ func (dn *CoreOSDaemon) applyOSChanges(mcDiff machineConfigDiff, oldConfig, newC
 	}
 
 	if dn.nodeWriter != nil {
-		dn.nodeWriter.Eventf(corev1.EventTypeNormal, "OSUpdateStaged", "Changes to OS staged")
+		// We send out the event OSUpdateStaged synchronously to ensure it is recorded.
+		// Remove this when we can ensure all events are sent before exiting.
+		ref, err := reference.GetReference(scheme.Scheme, dn.node)
```

Comment:
Are we just ignoring this error if the GetReference somehow fails? In that case the event would not send but we continue the update, so we should probably either explicitly fail or maybe not use GetReference here

Reply:
Yeah, I was ignoring the error. Removed GetReference anyway as it isn't needed when I can access the object directly like you stated in another comment.